src/module/skb_drop/event.rs

```diff
@@ -13,24 +20,59 @@ use crate::{
 // Skb drop event section. Same as the event from BPF, please keep in sync with
 // its BPF counterpart.
 #[event_section]
-#[repr(C, packed)]
 pub(crate) struct SkbDropEvent {
     /// Reason why a packet was freed/dropped. Only reported from specific
     /// functions. See `enum skb_drop_reason` in the kernel.
-    pub(crate) drop_reason: u32,
+    pub(crate) drop_reason: String,
 }
 
-unsafe impl Plain for SkbDropEvent {}
-
-#[derive(Default)]
 #[event_section_factory(SkbDropEvent)]
-pub(crate) struct SkbDropEventFactory {}
+pub(crate) struct SkbDropEventFactory {
+    reasons: HashMap<i32, String>,
+}
+
+impl SkbDropEventFactory {
+    pub(crate) fn new() -> Result<Self> {
+        let mut reasons = HashMap::new();
+
+        if let (btf, Type::Enum(r#enum)) =
+            inspect::btf_info()?.resolve_type_by_name("skb_drop_reason")?
+        {
+            for member in r#enum.members.iter() {
+                reasons.insert(
+                    member.val(),
+                    btf.resolve_name(member)?
+                        .trim_start_matches("SKB_")
+                        .trim_start_matches("DROP_REASON_")
+                        .replace('_', " ")
```

Comment:
This makes it human readable but more difficult difficult for programmatic matching (e.g: postprocessing).
I'm wondering if there's a way to keep the integer as data representation and only do this when marshaling/unmarshaling  or maybe having a type that is convertible to/from string...

We would not be able to use BTF in that case though...

Reply:
Maybe keeping the enum name and only making it human-readable for printing? Not ideal but I have no more ideas.

Reply:
> If you see a value in providing the raw value, we could have both in the event, "skb-drop":{"drop_reason":"tcp old data", "drop_reason_raw":42}}.

If the numeric value is not stable that won't add much.
I was thinking that if we implement a reason filter: `retis collect --reason-filter="TCP_OLD_DATA"` is sligthly nicer than `retis collect --reason-filter="tcp old data"` ... Maybe it's just because I'm thinking in reasons as error codes (in my mind I'd filter by EINVAL not "invalid argument")... not a super strong issue though